accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -554,19 +554,48 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
         self.set_age_to_future(entry, upsert_cached);
     }
 
+    /// Insert a cached entry into the accounts index
+    /// If the entry is already present, just mark dirty and set the age to the future
+    /// Code is required just for test for now, but will be used in future PRs so not putting in the test area
+    #[cfg(test)]
+    fn cache_entry_at_slot(&self, entry: &AccountMapEntry<T>, slot: Slot, account_info: T) {
+        let mut slot_list = entry.slot_list.write().unwrap();
+        if !slot_list
+            .iter()
+            .any(|(existing_slot, _)| *existing_slot == slot)
+        {
+            slot_list.push((slot, account_info));
+        }
+        entry.set_dirty(true);
+        self.set_age_to_future(entry, true);
+    }
+
     pub fn upsert(
         &self,
         pubkey: &Pubkey,
         new_value: PreAllocatedAccountMapEntry<T>,
         other_slot: Option<Slot>,
         reclaims: &mut SlotList<T>,
         reclaim: UpsertReclaim,
+    ) {
+        self.get_or_create_index_entry_for_pubkey(pubkey, |entry| {
+            self.update_slot_list_entry(entry, new_value, other_slot, reclaims, reclaim)
+        });
+    }
+
+    /// Gets an entry for `pubkey` and calls `callback` with it.
+    /// If the entry is not in the index, an empty entry will be created
+    /// If the entry is in the index, it will be returned as is
```

Comment:
nit: revise comments. 
```suggestion
    //// Gets an entry for `pubkey` and passes it to `callback`.
    /// If the entry is not in the index, an empty entry will be created and passed to `callback`.
    /// If the entry is in the index, it will be passed to `callback` as is.
```